src/VisualStudio/Core/Def/Implementation/LanguageClient/LanguageServerClientEventListener.cs

```diff
@@ -13,19 +19,58 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
     [ExportEventListener(WellKnownEventListeners.Workspace, WorkspaceKind.Host), Shared]
     internal class LanguageServerClientEventListener : IEventListener<object>
     {
+        private readonly LanguageServerClient _languageServerClient;
+        private readonly Lazy<ILanguageClientBroker> _languageClientBroker;
         private readonly TaskCompletionSource<object> _taskCompletionSource;
 
         public Task WorkspaceStarted => _taskCompletionSource.Task;
 
-        public LanguageServerClientEventListener()
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public LanguageServerClientEventListener(LanguageServerClient languageServerClient, Lazy<ILanguageClientBroker> languageClientBroker)
         {
-            _taskCompletionSource = new TaskCompletionSource<object>();
+            this._languageServerClient = languageServerClient;
+            this._languageClientBroker = languageClientBroker;
+            this._taskCompletionSource = new TaskCompletionSource<object>();
         }
 
+        /// <summary>
+        /// LSP clients do not necessarily know which language servers (and when) to activate as they are language agnostic.
+        /// We know we can provide <see cref="LanguageServerClient"/> as soon as the workspace is started, so tell the
+        /// <see cref="ILanguageClientBroker"/> to start loading it.
+        /// </summary>
         public void StartListening(Workspace workspace, object serviceOpt)
         {
             // mark that roslyn solution is added
-            _taskCompletionSource.SetResult(null);
+            this._taskCompletionSource.SetResult(null);
+
+            this.WorkspaceStarted.ContinueWith(_ =>
```

Comment:
😕 Why is this not just using `Task.Run`? It completes `WorkspaceStarted` before calling this line.

Reply:
I wasn't completely sure that workspace started would be completed here, but if it is then I can do that.

Should I be using jtf run, or task run?

Reply:
Task.Run is fine since you are calling `Forget` on the result.

Reply:
Ahh right yeah I'm not waiting for it so it won't deadlock.  Will update.